addons/account_payment/models/payment_transaction.py

```diff
@@ -88,7 +88,10 @@ def _compute_reference_prefix(self, provider_code, separator, **values):
             invoice_ids = self._fields['invoice_ids'].convert_to_cache(command_list, self)
             invoices = self.env['account.move'].browse(invoice_ids).exists()
             if len(invoices) == len(invoice_ids):  # All ids are valid
-                return separator.join(invoices.mapped('name'))
+                prefix = separator.join(invoices.mapped('name'))
+                if i_nr := values.get('installment_number'):
```

Comment:
How long can an installment number be (in chars)?

Reply:
It's usually just a single number, depending on the number of installments: 1, 2, 3 ...

Reply:
Fine then